examples/test-exec-doc.yml

```diff
@@ -0,0 +1,153 @@
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}             # Azure Client ID from secrets
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}             # Azure Tenant ID from secrets
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }} # Azure Subscription ID from secrets
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x' # Specify Python version 3.x
+
+    - name: Install dependencies
+      run: |
+        pip install --upgrade pip   # Upgrade pip
+        pip install PyGithub pyyaml # Install required Python packages
+
+    - name: Run tests and handle PR
+      env:
+        GITHUB_TOKEN: ${{ secrets.PAT }} # Personal Access Token from secrets
+      run: |
+        python <<EOF
+        import os
+        import subprocess
+        import shutil
+        import github
+
+        # Get GitHub token from environment variable
+        GITHUB_TOKEN = os.getenv('GITHUB_TOKEN')
+        # Authenticate with GitHub using PyGithub
+        g = github.Github(GITHUB_TOKEN)
+
+        # Check if 'ie' (Innovation Engine) is installed, install if not
+        if not shutil.which("ie"):
+            # Update package lists and install unzip
+            subprocess.run("sudo apt-get update && sudo apt-get install -y unzip", shell=True, check=True)
+            # Install Innovation Engine CLI
+            subprocess.run("curl -Lks https://aka.ms/install-ie | /bin/bash", shell=True, check=True)
+
+        # Retrieve repository information
+        result = subprocess.run(['git', 'config', '--get', 'remote.origin.url'], capture_output=True, text=True)
+        repo_url = result.stdout.strip()
+        parts = repo_url.split('/')
+        owner = parts[-2]                     # Repository owner
+        repo_name = parts[-1].replace('.git', '') # Repository name
+        # Access the repository using GitHub API
+        repo = g.get_repo(f"{owner}/{repo_name}")
+
+        # Get branch name from environment variable
+        BRANCH_NAME = os.environ['GITHUB_REF'].split('/')[-1]
+        if BRANCH_NAME != "main":
+            # Prepare to create or update a pull request to 'main'
+            PR_TITLE = f"Merge Exec Doc from {BRANCH_NAME} to main"
+            PR_BODY = "This PR attempts to publish your Exec Doc. Running tests on it now using Innovation Engine..."
+
+            # Check if a PR with the same title already exists
+            existing_pulls = repo.get_pulls(state='open', head=f"{owner}:{BRANCH_NAME}", base='main')
+            pr = None
+            for pull in existing_pulls:
+                if pull.title == PR_TITLE:
+                    pr = pull
+                    pr.edit(body=PR_BODY) # Update PR body if it exists
+                    break
+            if pr is None:
+                # Create a new pull request if none exists
+                pr = repo.create_pull(title=PR_TITLE, body=PR_BODY, head=f"{owner}:{BRANCH_NAME}", base='main')
```

Comment:
Shouldn't the branch that's being used for the push/PR be used for creating a new pull request? I'm not sure how the PR creation is working here

Reply:
it works from the same branch